readme.md
# Antistasi Amazons  

<img src="antistasi-amazons.jpg" alt="Antistasi Amazons illustration by Simon Griffee." width="100" align="right">

**Version 0.0.1** - 2020 June 12th  
*Antistasi Amazons* is currently based on [Antistasi Community Edition version 2.2.1][ascev].

[ascev]: https://github.com/official-antistasi-community/A3-Antistasi/releases/tag/2.2.1

***Antistasi Amazons*** is a basic modification (mod) of a dynamic long-term “Capture The Island” mission called ***[Antistasi][as]*** for the video game landscape orienteering and combat simulator *[Arma 3][a3]*. *Antistasi* was originally made by [Barbolani](https://www.youtube.com/user/barbolani/) and is now an open source [community-developed project][asc].

[a3]: https://www.bohemia.net/games/arma3
[as]: https://github.com/official-antistasi-community/A3-Antistasi
[asc]: https://github.com/official-antistasi-community/A3-Antistasi/graphs/contributors

## Features

- Uses two player-made client-side mods to add [women][w] to the game (the guerrillas you fight with), and lets you use a woman avatar.
- Adds a new rebel flag, introduction card, and map board card with the above illustration.
- Adds new briefing and overview text.
- Disables Arma’s [annoying][annoying] [stamina][st] mechanic.
- Adds a basic rifle and scope and removes one submachine gun from starting weapons.
- Uses standard *Arma 3* assets and its default **Altis** island map, a digital version of [Lemnos][lem].
- Sets saner mission parameter defaults to allow fast travel, reduce the number of weapons required for infinite unlocks to 10, and make enemy supply crate contents random.
- Playable in singleplayer or multiplayer modes.

## Installation

Please see [this blog post](https://hypertexthero.com/amazons) for installation instructions for both single and multiplayer modes.

[w]: https://en.wikipedia.org/wiki/Women_in_combat
[annoying]: https://community.bistudio.com/wiki/Arma_3_Stamina
[st]: https://www.washingtonpost.com/opinions/women-in-combat-roles-would-strengthen-the-military/2014/04/03/f0aeb140-bb50-11e3-9a05-c739f29ccb08_story.html 
[lem]: https://en.wikipedia.org/wiki/Lemnos

## Issues

- The hair of some of the female models in the [Max_women][fh] and [FEMAL3_Heads][mw] mods clip through hats and other head gear.
- When you take temporary control of an AI guerrilla and relinquish it after 60 seconds the AI character becomes your own avatar. 
- Bodies and voices.
  
[fh]: https://steamcommunity.com/workshop/filedetails/?id=374775446
[mw]: https://steamcommunity.com/workshop/filedetails/?id=1451755886

## Todo
  
- Add welcome sound to mission. First configure it in descriptions.ext, then [follow](https://community.bistudio.com/wiki/playSound3D) [these](https://steamcommunity.com/app/107410/discussions/0/143387886728301734/) [instructions](https://forums.bohemia.net/forums/topic/189498-solved-how-to-correctly-reference-a3-vanilla-sounds/).
- Add [random quotes](https://community.bistudio.com/wiki/BIS_fnc_selectRandom) function `_select1 = ["A", "B", "C", "D", "E"] call BIS_fnc_selectRandom;` and more updates to various mission texts.
- Eventually publish [on Steam](https://arma3.com/workshop101).

## Credits

- **[Antistasi](https://github.com/official-antistasi-community/A3-Antistasi)** by [Barbolani](https://www.youtube.com/user/barbolani/) and the [Antistasi Community Edition Team](https://github.com/official-antistasi-community/A3-Antistasi/graphs/contributors)
- **[Max_Women](https://steamcommunity.com/workshop/filedetails/?id=1451755886)** by [Maxjoiner](https://www.youtube.com/channel/UCPGvwaVtKMTa7EHrT-Ys4ww)
- **[FEMAL3_Heads](https://steamcommunity.com/workshop/filedetails/?id=374775446)** by [Zeealex](https://www.deviantart.com/zeealex)
- **[Arma 3](https://www.bohemia.net/games/arma3)** by [Bohemia Interactive](https://www.bohemia.net/)
- **Antistasi Amazons** basic mod by [Hypertexthero](https://hypertexthero.com)


Antistasi-Altis-Amazons.Altis/changelog.txt
0.1 24/04/2016 BETA

- Initial release


0.1.1 24/04/2016 BETA

- Fixed Persistent Save: It wont corrupt Altis saves.
- Corrected Maru respawning as Petros.
- Corrected punishment if SDK player kills a NATO player.
- Added an additional radio tower.
- Small bugfixes on fastrope.
- Flashlights should be used by SDK units (if no NV).
- Added unlocked optics for any SDK rifle users (no marksmen)
- Added Titan AA for militia if unlocked and skillcheck is ok.
- Added NV if unlocked too.
- Dogs dont make you lose Undercover if you are in a vehicle.
- SDK players no longer see NATO players markers.
- Engineer players can now repair vehicles.
- Officer players can now hack UAVs.
- NATO rebuilds towers again.
- Corrected refugees missions: they will only spawn on destroyed cities.
- Small +1 HR bonus on each Tax Report.
- Safer Convoy spawn.
- Fixed NATO player taxis.
- Added cleanser and punishment functions for NATO taxis.
- Added disconnected and connected functions for NATO players.
- NATO players no longer see SDK markers.
- NATO players receive notifications on certain SDK actions (supplies delivered on cities etc.).
- Markers update correctly on Pers. Save.
- Increased a bit the distance to land for helis as they are crashing a lot with trees.
- AI won't attack other islands targets except Airports on major attacks.
- CSAT Starts controlling the whole NW island.
- Fixed bug on initial HQ placement.
- Greatly improved safety and speed on despawning procedures.
- Halved Quadbike cost.
- Gendarmes and FIA will use flashlights.


0.1.2 24/04/2016 BETA

- NEW FEATURE: Commanders can clear the nearby forest in order to have more space for their vehicles. Persistent Save integrated.
- Evac Refugees can happen in NATO cities with substantial differences between NATO and CSAT mission types.
- So, removed the +1 HR bonus.
- Corrected heavy bugs on CSAT punishments.
- CSAT Punish defending civvies number spawned independenty of the spawn rate.
- Map info shows "NONE" when city is supported by a radio tower in a CSAT outpost.
- Corrected: Lose Undercover if you kill an enemy running through him with a vehicle.
- Corrected major bug in Traitor missions.
- City support changes on each kill no matter if it's SDK or not.
- Solved undercover issues when players FFV.
- Changed the name of "Airports" to "Airbases".
- NATO Garrisons increased a bit if they are guarding a RT.
- Tweaked bonuses for defeating a CSAT attack depending on several params.
- NATO players will receive major attacks tasks.
- Earplugs script added to NATO players.
- NAPALM destroys forests.
- Units respawn in their proper respawn point once they finish their punishment.
- Corrected RT Rebuild bug.
- Prestige values saved correctly.


0.1.2.3 26/08/2016 BETA

- Tweaked here and there Undercover and Revive to prevent some bugs MP related.
- Medics are now truly medics.
- Added ragdolling and nice effects for uncosnscious state.
- Added 30 secs respawn delay.
- When players respawn they will remain in control of their IEDs planted before dying.
- Tweaked bonuses when central airport is attacked.
- Corrected clear forest in MP.
- Engineers will be able to disarm bombs.
- Ammo and repair trucks will spawn on near roads on their respective missions.
- Deads by NAPALM burn a bit like campfires for some time.
- NATO WONT send a QRF to defend a city under attack if it's SDK.


0.1.2.4 27/08/2016

- IMPORTANT: Loosing condition: 1/3 of the total population is massacred by CSAT. Number will be shown on "Map Info".
- IMPORTANT: R key to respawn.
- IMPORTANT: Win conditions: more then half of the population supports SDK and SDK owns all the airports.
- Improved AI vehicle unstuck functions.
- "FIA" replaced by "SDK" in a few texts.
- Changed model for SDK crewmans (FIA model, as BIS does not give us a proper unarmed SDK soldier)
- Arty/Bomb run markers are now local for the SDK Commander.
- MAP info will report enemy garrison status.
- AI will try to avoid NAPALM.
- Changed default unlocked backpack.
- Randomised grunts rifles.
- Added Tanoa NATO drones as attack helis
- Added backpacks to the possible AI ammocrate loadout.
- Added temporal inmunity for Maru just in case he dies because of NAPALM.
- Changed speed waypoint params for Convoys by suggestion of one player. Under testing.
- Airstrikes in QRF were accidentally half removed.
- Unconscious units will disembark from vehicles before reaching unconscious state.
- AIs from sidemissions won't make anything spawn.
- Fixed when commander disconnects and despawns his AI + vehicles despite he's got players in his group.


0.3 28/08/2016

- IMPORTANT: jungles are no longer a sure safe position. SpecOp patrols cover the area if AI feels there is insurgency nearby. If SDK kills all the patrols, they will respawn in the same way roadblocks do. Sometimes, AI will decide to reinforce the area with mines.
- Traits on engineers and medics applied to SDK AI too.
- Dead AI teammates will get the "Revive" action deleted.
- Refugees Evac: On NATO cities a Police car will spawn nearby to help the player on his search.
- Convoys wont spawn during major attacks.
- AI wont attack refugees when delivered at HQ.
- Crew of downed AI Air vehicles will auto-die to reduce spawned units in the field.
- Counter for major attacks will never be less than zero.
- Added a lot predefined placed roadblocks.
- Fixed delay when Petros was killed.
- Few pesky "FIA" here and there.
- Some anti lag measures when AI attacks an Airbase.


0.3.1 28/08/2016

- Greatly improved spawning garrisons with all the cache related.
- Tasks on enemy or friendly major attacks no matter if you are the target or not.
- Tweaked a lot spawning numbers on AI major attacks.
- Tweaked a lot simulation enablement on several situations to improve FPS.

Antistasi-Altis-Amazons.Altis/defines.hpp
//Common Macros
#ifndef QUOTE
#define QUOTE(var1) #var1
#endif

//Other Includes
#include "functions\Garage\defineGarage.inc"

// Control types
#define CT_STATIC           0
#define CT_BUTTON           1
#define CT_EDIT             2
#define CT_SLIDER           3
#define CT_COMBO            4
#define CT_LISTBOX          5
#define CT_TOOLBOX          6
#define CT_CHECKBOXES       7
#define CT_PROGRESS         8
#define CT_HTML             9
#define CT_STATIC_SKEW      10
#define CT_ACTIVETEXT       11
#define CT_TREE             12
#define CT_STRUCTURED_TEXT  13
#define CT_CONTEXT_MENU     14
#define CT_CONTROLS_GROUP   15
#define CT_SHORTCUTBUTTON   16
#define CT_XKEYDESC         40
#define CT_XBUTTON          41
#define CT_XLISTBOX         42
#define CT_XSLIDER          43
#define CT_XCOMBO           44
#define CT_ANIMATED_TEXTURE 45
#define CT_OBJECT           80
#define CT_OBJECT_ZOOM      81
#define CT_OBJECT_CONTAINER 82
#define CT_OBJECT_CONT_ANIM 83
#define CT_LINEBREAK        98
#define CT_USER             99
#define CT_MAP              100
#define CT_MAP_MAIN         101
#define CT_LISTNBOX         102

// Static styles
#define ST_POS            0x0F
#define ST_HPOS           0x03
#define ST_VPOS           0x0C
#define ST_LEFT           0x00
#define ST_RIGHT          0x01
#define ST_CENTER         0x02
#define ST_DOWN           0x04
#define ST_UP             0x08
#define ST_VCENTER        0x0C
#define ST_GROUP_BOX       96
#define ST_GROUP_BOX2      112
#define ST_ROUNDED_CORNER  ST_GROUP_BOX + ST_CENTER
#define ST_ROUNDED_CORNER2 ST_GROUP_BOX2 + ST_CENTER

#define ST_TYPE           0xF0
#define ST_SINGLE         0x00
#define ST_MULTI          0x10
#define ST_TITLE_BAR      0x20
#define ST_PICTURE        0x30
#define ST_FRAME          0x40
#define ST_BACKGROUND     0x50
#define ST_GROUP_BOX      0x60
#define ST_GROUP_BOX2     0x70
#define ST_HUD_BACKGROUND 0x80
#define ST_TILE_PICTURE   0x90
#define ST_WITH_RECT      0xA0
#define ST_LINE           0xB0

#define ST_SHADOW         0x100
#define ST_NO_RECT        0x200
#define ST_KEEP_ASPECT_RATIO  0x800

#define ST_TITLE          ST_TITLE_BAR + ST_CENTER

// Slider styles
#define SL_DIR            0x400
#define SL_VERT           0
#define SL_HORZ           0x400

#define SL_TEXTURES       0x10

// progress bar
#define ST_VERTICAL       0x01
#define ST_HORIZONTAL     0

// Listbox styles
#define LB_TEXTURES       0x10
#define LB_MULTI          0x20

// Tree styles
#define TR_SHOWROOT       1
#define TR_AUTOCOLLAPSE   2

// MessageBox styles
#define MB_BUTTON_OK      1
#define MB_BUTTON_CANCEL  2
#define MB_BUTTON_USER    4


////////////////
//Base Classes//
////////////////

class RscText
{
    access = 0;
    idc = -1;
    type = CT_STATIC;
    style = ST_MULTI;
    linespacing = 1;
    colorBackground[] = {0,0,0,0};
    colorText[] = {1,1,1,.5};
    text = "";
    shadow = 2;
    font = "PuristaMedium";
    SizeEx = 0.02300;
    fixedWidth = 0;
    x = 0;
    y = 0;
    h = 0;
    w = 0;

};

class RscPicture
{
    access = 0;
    idc = -1;
    type = CT_STATIC;
    style = ST_PICTURE;
    colorBackground[] = {0,0,0,0};
    colorText[] = {1,1,1,1};
    font = "PuristaMedium";
    sizeEx = 0;
    lineSpacing = 0;
    text = "";
    fixedWidth = 0;
    shadow = 0;
    x = 0;
    y = 0;
    w = 0.2;
    h = 0.15;
};

class RscButton
{

access = 0;
    type = CT_BUTTON;
    text = "";
    colorText[] = {0.73,0,0,1};
    colorDisabled[] = {0.4,0.4,0.4,0};
    colorBackground[] =  {0.247,0.243,0.243,1};
    colorBackgroundDisabled[] = {0,0.0,0};
    colorBackgroundActive[] = {0.247,0.243,0.243,1};
    colorFocused[] = {0.247,0.243,0.243,1};
    colorShadow[] = {0.023529,0,0.0313725,1};
    colorBorder[] = {0.023529,0,0.0313725,1};
    soundEnter[] = {"\A3\ui_f\data\sound\RscButton\soundEnter",0.09,1};
    soundPush[] = {"\A3\ui_f\data\sound\RscButton\soundPush",0.09,1};
    soundClick[] = {"\A3\ui_f\data\sound\RscButton\soundClick",0.09,1};
    soundEscape[] = {"\A3\ui_f\data\sound\RscButton\soundEscape",0.09,1};
    style = 2;
    x = 0;
    y = 0;
    w = 0.055589;
    h = 0.039216;
    shadow = 2;
    font = "PuristaMedium";
    sizeEx = 0.05;
    offsetX = 0.003;
    offsetY = 0.003;
    offsetPressedX = 0.002;
    offsetPressedY = 0.002;
    borderSize = 0;
    onMouseEnter = "(_this select 0) ctrlSetTextColor [1,0.969,0,1]";
    onMouseExit = "(_this select 0) ctrlSetTextColor [0.73,0,0,1]";
};

class RscFrame
{
    type = CT_STATIC;
    idc = -1;
    style = ST_FRAME;
    shadow = 2;
    colorBackground[] = {1,1,1,1};//{1,1,1,1}
    colorText[] = {1,1,1,0.9};
    font = "PuristaMedium";
    sizeEx = 0.03;
    text = "";
};
class BOX
{
   type = CT_STATIC;
    idc = -1;
    style = ST_CENTER;
    shadow = 2;
    colorText[] = {1,1,1,1};
    font = "PuristaMedium";
    sizeEx = 0.02;
    colorBackground[] = { 0.2,0.2,0.2, 0.9 };
    text = "";
};
/*
class ScrollBar
 {
  arrowEmpty = "#(argb,8,8,3)color(1,1,1,1)";
  arrowFull = "#(argb,8,8,3)color(1,1,1,1)";
  border = "#(argb,8,8,3)color(1,1,1,1)";
  color[] = {1,1,1,0.6};
  colorActive[] = {1,1,1,1};
  colorDisabled[] = {1,1,1,0.3};
  thumb = "#(argb,8,8,3)color(1,1,1,1)";
 };
*/
class RscListBox
{
     access = 0;
     type = 5;
     style = 0;
     w = 0.4;
     h = 0.4;
     font = "TahomaB";
     sizeEx = 0.04;
     rowHeight = 0;
     colorText[] = {1,1,1,1};
     colorScrollbar[] = {1,1,1,1};
     colorSelect[] = {0,0,0,1};
     colorSelect2[] = {1,0.5,0,1};
     colorSelectBackground[] = {0.6,0.6,0.6,1};
     colorSelectBackground2[] = {0.2,0.2,0.2,1};
     colorBackground[] = {0.2,0.2,0.2,0.9};
     maxHistoryDelay = 1.0;
     soundSelect[] = {"",0.1,1};
     period = 1;
     autoScrollSpeed = -1;
     autoScrollDelay = 5;
     autoScrollRewind = 0;
     arrowEmpty = "#(argb,8,8,3)color(1,1,1,1)";
     arrowFull = "#(argb,8,8,3)color(1,1,1,1)";
     shadow = 0;
     class ListScrollBar// : ScrollBar //ListScrollBar is class name required for Arma 3
     {
          color[] = {1,1,1,0.6};
          colorActive[] = {1,1,1,1};
          colorDisabled[] = {1,1,1,0.3};
          thumb = "#(argb,8,8,3)color(1,1,1,1)";
          arrowEmpty = "#(argb,8,8,3)color(1,1,1,1)";
          arrowFull = "#(argb,8,8,3)color(1,1,1,1)";
          border = "#(argb,8,8,3)color(1,1,1,1)";
          shadow = 0;
     };
};


Antistasi-Altis-Amazons.Altis/MissionDescription/CfgIdentities.hpp
class CfgIdentities
{
  class protagonista
  {
    name = "Didion";
    nameSound = "Didion";
    face="max_female4";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.5;
  };
  class friendlyX // Petros
  {
    name = "Petros";
    nameSound = "Petros";
    face="max_female15";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.4;
  };
  class Anthis
  {
    name = "Amos";
    nameSound = "Amos";
    face="max_female5";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.6;
  };
  class Costa
  {
    name = "Atwood";
    nameSound = "Atwood";
    face="max_female6";
    glasses="None";
    speaker="Male02GRE";
    pitch=1.5;
  };
  class Dimitirou
  {
    name = "Lessing";
    nameSound = "Lessing";
    face="Arnott";
    glasses="None";
    speaker="Male03GRE";
    pitch=1.4;
  };
  class Elias
  {
    name = "Ziadeh";
    nameSound = "Ziadeh";
    face="max_female9";
    glasses="None";
    speaker="Male04GRE";
    pitch=1.3;
  };
  class Gekas
  {
    name = "Enchi";
    nameSound = "Enchi";
    face="Oakes";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.7;
  };
  class Kouris
  {
    name = "Chang";
    nameSound = "Chang";
    face="Mason";
    glasses="None";
    speaker="Male02GRE";
    pitch=1.6;
  };
  class Leventis
  {
    name = "Shelley";
    nameSound = "Shelley";
    face="max_female7";
    glasses="None";
    speaker="Male03GRE";
    pitch=1.6;
  };
  class Markos
  {
    name = "Woolf";
    nameSound = "Woolf";
    face="max_female8";
    glasses="None";
    speaker="Male04GRE";
    pitch=1.2;
  };
  class Nikas
  {
    name = "Angelou";
    nameSound = "Angelou";
    face="max_female9";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.3;
  };
  class Nicolo
  {
    name = "Morrison";
    nameSound = "Morrison";
    face="max_female17";
    glasses="None";
    speaker="Male02GRE";
    pitch=1.2;
  };
  class Panas
  {
    name = "Adichie";
    nameSound = "Adichie";
    face="max_female16";
    glasses="None";
    speaker="Male03GRE";
    pitch=1.4;
  };
  class Rosi
  {
    name = "de Beauvoir";
    nameSound = "de Beauvoir";
    face="max_female10";
    glasses="None";
    speaker="Male04GRE";
    pitch=1.4;
  };
  class Samaras
  {
    name = "Le Guin";
    nameSound = "Le Guin";
    face="max_female11";
    glasses="None";
    speaker="Male01GRE";
    pitch=1.5;
  };
  class Thanos
  {
    name = "Brontë";
    nameSound = "Brontë";
    face="max_female12";
    glasses="None";
    speaker="Male02GRE";
    pitch=1.5;
  };
  class Vega
  {
    name = "Eliot";
    nameSound = "Eliot";
    face="max_female13";
    glasses="None";
    speaker="Male03GRE";
    pitch=1.4;
  };
};

Antistasi-Altis-Amazons.Altis/MissionDescription/params.hpp
class Params
{
     class loadSave
     {
          title = "Load last Persistent Save";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class gameMode
     {
          title = "Game Mode";
          values[] = {1,2,3,4};
          texts[] = {"Reb vs Gov vs Inv","Reb vs Gov & Inv","Reb vs Gov","Reb vs Inv"};
          default = 1;
     };
     class autoSave
     {
          title = "Enable Autosave (every hour)";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class membership
     {
          title = "Enable Server Membership";
          texts[] = {"Yes","No"};
          values[] = {1,0};
          default = 1;
     };
     class switchComm
     {
          title = "Enable Commander Switch (highest ranked player)";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class tkPunish
     {
          title = "Enable Teamkill Punish";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class mRadius
     {
          title = "Distance from HQ for Sidemissions";
          values[] = {2000,4000,6000,8000,10000,12000};
          default = 4000;
     };
     class allowPvP
     {
          title = "Allow PvP Slots";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class pMarkers
     {
          title = "Allow Friendly Player Markers";
          values[] = {1,0};
          texts[] = {"Yes","No"};
          default = 1;
     };
     class AISkill
     {
          title = "Mission Difficulty";
          values[] = {1,2,3};
          texts[] = {"Easy","Normal","Hard"};
          default = 2;
     };
     class unlockItem
     {
          title = "Number of the same weapons required to unlock";
          values[] = {10,25,40};
          default = 10;
     };
     class memberOnlyMagLimit
     {
          title = "Number of magazines needed for guests to be able to use them";
          values[] = {10,20,30,40,50,60};
          default = 40;
     };
     class civTraffic
     {
          title = "Rate of Civ Traffic";
          values[] = {0,1,2,3};
          texts[] = {"None","Low","Medium","JAM"};
          default = 1;
     };
     class memberSlots
     {
          title = "Percentage of Reserved Slots for Members";
          values[] = {0,20,40,60,80,100};
          texts[] = {"None","20%","40%","60%","80%","All"};
          default = 20;
     };
     class memberDistance
     {
          title = "Max distance non members can be from the closest member or HQ (they will be teleported to HQ after some timeout)";
          values[] = {4000,5000,6000,7000,8000,16000};
          texts[] = {"4 Kmts","5 Kmts","6 Kmts","7 Kmts","8 Kmts","Unlimited"};
          default = 5000;
     };
	 class allowMembersFactionGarageAccess
     {
          title = "Allow members to access the faction garage";
          texts[] = {"Yes", "No"};
          values[] = {1,0};
          default = 1;
     };
     class allowFT
     {
          title = "Limited Fast Travel";
          values[] = {0,1};
          texts[] = {"No","Yes"};
          default = 0;
     };
     class napalmEnabled
     {
          title = "Enable Napalm Bombing for AI";
          values[] = {0,1};
          texts[] = {"No","Yes"};
          default = 0;
     };
     class teamSwitchDelay
     {
          title = "Delay After Leaving Before a Player Can Join Another Team";
          values[] = {0, 900, 1800, 3600};
          texts[] = {"No delay","15 minutes","30 minutes","60 minutes"};
          default = 900;
     };
     class unlockedUnlimitedAmmo
     {
          title = "Do Unlocked Weapons Automatically Unlock Their Standard Magazine?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class allowGuidedLaunchers
     {
          title = "Should Guided Launchers become unlocked?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 1;
     };
     class allowUnlockedExplosives
     {
          title = "Should Explosives become unlocked?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 1;
     };
     class startWithLongRangeRadio
     {
          title = "[TFAR] Start with Long Range Radio?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 1;
     };
     class Spacer10
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class Kart
     {
          title = "Allow Items and Vehicles from Karts DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Mark
     {
          title = "Allow Items and Vehicles from Marksman DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Heli
     {
          title = "Allow Items and Vehicles from Heli DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Expansion
     {
          title = "Allow Items and Vehicles from Apex DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Jets
     {
          title = "Allow Items and Vehicles from Jets DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Orange
     {
          title = "Allow Items and Vehicles from Laws of War DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Tanks
     {
          title = "Allow Items and Vehicles from Tanks DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class GlobMob
     {
          title = "Allow Items and Vehicles from Global Mobilization DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Enoch
     {
          title = "Allow Items and Vehicles from Contact DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class OfficialMod
     {
          title = "Allow ADR-97 DLC?";
          values[] = {1,0};
          texts[] =  {"Yes","No"};
          default = 0;
     };
     class Spacer0
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class AdvancedParams
     {
          title = "ADVANCED USERS ONLY";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
	 class LogLevel
	 {
		  title = "Logging Level (Amount of detail in .rpt file)";
		  values[] = {1,2,3};
		  texts[] = {"Error", "Info", "Debug"};
		  default = 3;
	 };
     class CrateOptions
     {
          title = "LOOT CRATE OPTIONS";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
	 class truelyRandomCrates
	 {
		  title = "[Experimental] Truely Random Crates: Remove all balance checks from Crates";
		  values[] = {0, 1};
		  texts[] = {"False", "True"};
		  default = 1;
	 };
	 class cratePlayerScaling
	 {
		title = "Decrease loot quantity as player count increases? (Yes is recommended for balance reasons)";
		values[] = {0, 1};
		texts[] = {"False", "True"};
		default = 1;
	 };
     class crateWepTypeMax
     {
          title = "Maximum Weapon Types in Crates";
          values[] = {0,2,4,6,8,12,16};
          texts[] = {"1","3","5","7","9","13","17"};
          default = 9;
     };
     class crateWepNumMax
     {
          title = "Maximum Weapon Quantity in Crates";
          values[] = {0,1,3,5,8,10,15};
          texts[] = {"None","1","3","5","8","10","15"};
          default = 8;
     };
     class Spacer1
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateItemTypeMax
     {
          title = "Maximum Item Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 4;
     };
     class crateItemNumMax
     {
          title = "Maximum Item Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 5;
     };
     class Spacer2
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateAmmoTypeMax
     {
          title = "Maximum Ammo Types in Crates";
          values[] = {0,2,4,6,9,14,19};
          texts[] = {"1","3","5","7","10","15","20"};
          default = 6;
     };
     class crateAmmoNumMax
     {
          title = "Maximum Ammo Quantity in Crates";
          values[] = {0,1,3,5,10,15,20,25,30};
          texts[] = {"None","1","3","5","10","15","20","25","30"};
          default = 20;
     };
     class Spacer3
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateExplosiveTypeMax
     {
          title = "Maximum Explosive Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 2;
     };
     class crateExplosiveNumMax
     {
          title = "Maximum Explosive Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 5;
     };
     class Spacer4
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateAttachmentTypeMax
     {
          title = "Maximum Attachment Types in Crates";
          values[] = {0,2,4,6,9,12,15,19};
          texts[] = {"1","3","5","7","10","13","16","20"};
          default = 6;
     };
     class crateAttachmentNumMax
     {
          title = "Maximum Attachment Quantity in Crates";
          values[] = {0,1,3,5,7,10,15,20,30};
          texts[] = {"None","1","3","5","7","10","15","20","30"};
          default = 15;
     };
     class Spacer5
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateBackpackTypeMax
     {
          title = "Maximum Backpack Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 0;
     };
     class crateBackpackNumMax
     {
          title = "Maximum Backpack Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 3;
     };
     class Spacer6
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateVestTypeMax
     {
          title = "Maximum Vest Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 0;
     };
     class crateVestNumMax
     {
          title = "Maximum Vest Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 0;
     };
     class Spacer7
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateHelmetTypeMax
     {
          title = "Maximum Helmet Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 0;
     };
     class crateHelmetNumMax
     {
          title = "Maximum Helmet Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 0;
     };
     class Spacer8
     {
          title = "";
          values[] = {""};
          texts[] = {""};
          default = "";
     };
     class crateDeviceTypeMax
     {
          title = "Maximum Device Backpack Types in Crates";
          values[] = {0,2,4,9};
          texts[] = {"1","3","5","10"};
          default = 2;
     };
     class crateDeviceNumMax
     {
          title = "Maximum Device Backpack Quantity in Crates";
          values[] = {0,1,3,5,10,15};
          texts[] = {"None","1","3","5","10","15"};
          default = 3;
     };
};
